lib/core/sessions.js

```diff
@@ -188,6 +191,11 @@ class ClientSession extends EventEmitter {
       throw new MongoError('Transaction already in progress');
     }
 
+    const maxWireVersion = this.topology.lastIsMaster().maxWireVersion;
+    if (isSharded(this.topology) || maxWireVersion < MAX_FOR_TRANSACTIONS) {
+      throw new MongoError('Transactions are not supported.');
```

Comment:
is this the prescribed error message? Perhaps "Transactions are not supported on sharded clusters in MongoDB < 4.0" or something along those lines.

Reply:
I agree with @mbroadst's comment, except I'd say "MongoDB < 4.2".

Reply:
doh, yeah of course :)